src/node/async_hooks.ts

```diff
@@ -7,7 +7,105 @@ import { default as async_hooks } from 'node-internal:async_hooks';
 
 export const { AsyncLocalStorage, AsyncResource } = async_hooks;
 
+export const asyncWrapProviders: Record<string, number> = {
+  NONE: 0,
+  DIRHANDLE: 1,
+  DNSCHANNEL: 2,
+  ELDHISTOGRAM: 3,
+  FILEHANDLE: 4,
+  FILEHANDLECLOSEREQ: 5,
+  BLOBREADER: 6,
+  FSEVENTWRAP: 7,
+  FSREQCALLBACK: 8,
+  FSREQPROMISE: 9,
+  GETADDRINFOREQWRAP: 10,
+  GETNAMEINFOREQWRAP: 11,
+  HEAPSNAPSHOT: 12,
+  HTTP2SESSION: 13,
+  HTTP2STREAM: 14,
+  HTTP2PING: 15,
+  HTTP2SETTINGS: 16,
+  HTTPINCOMINGMESSAGE: 17,
+  HTTPCLIENTREQUEST: 18,
+  JSSTREAM: 19,
+  JSUDPWRAP: 20,
+  MESSAGEPORT: 21,
+  PIPECONNECTWRAP: 22,
+  PIPESERVERWRAP: 23,
+  PIPEWRAP: 24,
+  PROCESSWRAP: 25,
+  PROMISE: 26,
+  QUERYWRAP: 27,
+  QUIC_ENDPOINT: 28,
+  QUIC_LOGSTREAM: 29,
+  QUIC_PACKET: 30,
+  QUIC_SESSION: 31,
+  QUIC_STREAM: 32,
+  QUIC_UDP: 33,
+  SHUTDOWNWRAP: 34,
+  SIGNALWRAP: 35,
+  STATWATCHER: 36,
+  STREAMPIPE: 37,
+  TCPCONNECTWRAP: 38,
+  TCPSERVERWRAP: 39,
+  TCPWRAP: 40,
+  TTYWRAP: 41,
+  UDPSENDWRAP: 42,
+  UDPWRAP: 43,
+  SIGINTWATCHDOG: 44,
+  WORKER: 45,
+  WORKERHEAPSNAPSHOT: 46,
+  WRITEWRAP: 47,
+  ZLIB: 48,
+  CHECKPRIMEREQUEST: 49,
+  PBKDF2REQUEST: 50,
+  KEYPAIRGENREQUEST: 51,
+  KEYGENREQUEST: 52,
+  KEYEXPORTREQUEST: 53,
+  CIPHERREQUEST: 54,
+  DERIVEBITSREQUEST: 55,
+  HASHREQUEST: 56,
+  RANDOMBYTESREQUEST: 57,
+  RANDOMPRIMEREQUEST: 58,
+  SCRYPTREQUEST: 59,
+  SIGNREQUEST: 60,
+  TLSWRAP: 61,
+  VERIFYREQUEST: 62,
+};
+
+export function createHook(): void {
+  throw new Error('createHook is not implemented');
```

Comment:
If we're going to go ahead and add stubs like this directly in the runtime, let's make these proper Node.js ERR codes using `ERR_METHOD_NOT_IMPLEMENTED`. Also, it may be worth expanding the error message for these in particular to indicate that they are not only not implemented but there is no intention of implementing these in order to better discourage folks from opening issues about them requesting them

Reply:
I converted them to ERR_METHOD_NOT_IMPLEMENTED but IMHO all of them can be implemented in the future. I don't see any technical blockers, if I'm not missing something.